src/api/pool.go

```diff
@@ -0,0 +1,109 @@
+package api
+
+import (
+	"context"
+	"github.com/go-resty/resty/v2"
+	"golang.org/x/sync/errgroup"
+	"runtime"
+	"sync"
+)
+
+type Pool struct {
+	client          *Client         // http client
+	ctx             context.Context // context of the parallel work
+	workers         *errgroup.Group // error group -> if one worker fails, all will be stopped
+	counter         sync.WaitGroup  // detect when all requests are processed
+	sendersCount    int
+	processorsCount int
+	processor       func(pool *Pool, response *resty.Response) error
+	requests        chan *resty.Request
+	responses       chan *resty.Response
+}
+
+func (c *Client) NewPool(processor func(pool *Pool, response *resty.Response) error) *Pool {
+	workers, ctx := errgroup.WithContext(c.parentCtx)
+	return &Pool{
+		client:          c,
+		ctx:             ctx,
+		workers:         workers,
+		counter:         sync.WaitGroup{},
+		sendersCount:    MaxIdleConns,
+		processorsCount: runtime.NumCPU(),
+		processor:       processor,
+		requests:        make(chan *resty.Request, 500),
+		responses:       make(chan *resty.Response),
+	}
+}
+
+// R creates request
+func (p *Pool) R(method string, url string) *resty.Request {
+	r := p.client.R().SetContext(p.ctx)
```

Comment:
Do Requestu sa pridava `context` to znamena, ze ak nastane chyba, tak sa jednoducho vsetko zastavi.